compute_endpoint/tests/integration/endpoint/executors/mock_executors.py

```diff
@@ -27,9 +31,19 @@ def start(
         self.endpoint_id = endpoint_id
         self.run_dir = run_dir
 
-    def submit(self, task_id: uuid.UUID, packed_task: bytes):
+    def submit(
+        self,
+        task_id: uuid.UUID,
+        packed_task: bytes,
+        resource_specification: t.Dict,
+    ):
         task: Task = messagepack.unpack(packed_task)
         res = Result(task_id=task_id, data=task.task_buffer)
+
+        # This is a hack to trigger an InvalidResourceSpecification
+        if "BAD_KEY" in resource_specification:
+            raise InvalidResourceSpecification("BAD_KEY")
```

Comment:
Rather than relying on `"BAD_KEY"`, is there a single point of authority within Parsl that we can reference for known good keys?

Reply:
Parsl's [validate_resource_spec](https://github.com/Parsl/parsl/blob/master/parsl/executors/high_throughput/mpi_prefix_composer.py#L44) has an internal datastructure holding the valid keys. Either way, the test is checking whether passing a bad ` resource_specification` which triggers `InvalidResourceSpefication` as `validate_resource_spec` would, is captured and reported properly. I think a generic "BAD_KEY" string works for this purpose.